### ISSUE TITLE
miss "subquery returns more than 1 row" check when MaxChunkSize=1

### ISSUE THREAD

**Issue by lysu** [pingcap/tidb#7333]
This issue will be fixed with #7141

1. What did you do?

```
create table t(id int primary key, v int);
insert into t values(1, 1), (2, 2), (3, 3);
set @@tidb_max_chunk_size=1;
select (select t.id from t where t.id = t.v and t.v != s.id) from t s;
```

2. What did you expect to see?

```
ERROR 1105 (HY000): subquery returns more than 1 row
```

3. What did you see instead?

```
+-------------------------------------------------------+
| (select t.id from t where t.id = t.v and t.v != s.id) |
+-------------------------------------------------------+
|                                                     2 |
|                                                     1 |
|                                                     1 |
+-------------------------------------------------------+
3 rows in set (0.01 sec)

```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?

```
Release Version: None
Git Commit Hash: None
Git Branch: None
UTC Build Time: None
GoVersion: None
Race Enabled: false
TiKV Min Version: 2.1.0-alpha.1-ff3dd160846b7d1aed9079c389fc188f7f5ea13e
```

**Comment by zz-jason** [comment]
@lysu We should fix this issue in an individual PR and cherrypick that PR to release-2.0

**Comment by lysu** [comment]
fixed by #7375
